src/material.h

```diff
@@ -75,4 +77,13 @@ class material_type
         const mat_burn_data &burn_data( size_t intensity ) const;
 };
 
+namespace materials
```

Comment:
YAGNI is barely relevant here. There's no "new stuff", no new level of indirection, and hence no additional cost. There's the "existing stuff" that should be placed in the right place. By placing it where it shouldn't be I'd make a "needless direction".